src/nats.h

```diff
@@ -1206,6 +1206,79 @@ typedef struct jsFetchRequest
 
 } jsFetchRequest;
 
+/** \brief Callback used to indicate that the work of js_PullSubscribeAsync is
+ * done.
+ *
+ * @param nc - Connection to the NATS server
+ * @param sub - Subscription being used
+ * @param s - Completion status code
+ * - `NATS_OK` - should never happen here!
+ * - `NATS_TIMEOUT` indicates that the fetch has reached its lifetime expiration
+ *   time, or had NoWait set and there are no more messages.
+ * - `NATS_NOT_FOUND` is returned when the server has no messages to deliver at
+ *   the beginning of a specific request. It may be returned for NoWait
+ *   subscriptions, effectively the same meaning as NATS_TIMEOUT - early
+ *   termination for NoWait.
+ * - `NATS_MAX_DELIVERED_MSGS` indicates that lifetime `Batch` message limit has
+ *   been reached.
+ * - `NATS_LIMIT_REACHED` is returned when the lifetime byte limit is reached.
+ * - Other status values represent error conditions.
+ * @param closure completeClosure that was passed to js_PullSubscribeAsync
+ *
+ * @see js_PullSubscribeAsync
+ */
+typedef void (*natsFetchCompleteHandler)(natsConnection *nc, natsSubscription *sub, natsStatus s, void *closure);
+
+/** \brief Callback used to customize flow control for js_PullSubscribeAsync.
+ *
+ * The library will invoke this callback when it may be time to request more
+ * messages from the server.
+ *
+ * @return true to fetch more, false to skip. If true, @messages and @maxBytes
+ * should be set to the number of messages and max bytes to fetch.
+ *
+ * @see js_PullSubscribeAsync
+ */
+typedef bool (*natsFetchNextHandler)(int *messages, int64_t *bytes, natsSubscription *sub, void *closure);
+
+typedef struct jsOptionsPullSubscribeAsync
```

Comment:
You don't need to make it a separate typedef. Put it back the way it was in `jsOptions`:
```
struct jsOptionsPullSubscribeAsync
        {
                // Lifetime of the subscription (completes when any one of the
                // targets is reached).
                int64_t                 Timeout; // in milliseconds
...
} PullSubscribeAsync;
```
Then, in natsp.h:
```
typedef struct __jsFetch
{
    struct jsOptionsPullSubscribeAsync opts;
...
```
That will work.

Reply:
Oh nice, I didn't realize you could re-use those nested struct definitions.